src/Native/System.Native/pal_errno.cpp

```diff
@@ -299,7 +303,11 @@ extern "C" int32_t SystemNative_ConvertErrorPalToPlatform(Error error)
         case PAL_ENOTEMPTY:
             return ENOTEMPTY;
         case PAL_ENOTRECOVERABLE:
+#if HAVE_ENOTRECOVERABLE
             return ENOTRECOVERABLE;
+#else
+            return -1;
+#endif
```

Comment:
The error was due to clang switch warning, which is fixed up by 891635a. Now the default falls through the assert.